source/h2_stream.c

```diff
@@ -270,11 +270,9 @@ static void s_stream_cross_thread_work_task(struct aws_channel_task *task, void
         struct aws_h2_frame *frame = AWS_CONTAINER_OF(node, struct aws_h2_frame, node);
         aws_h2_connection_enqueue_outgoing_frame(connection, frame);
     }
-    /* We already enqueued the window_update frame, just apply the change and let our peer check this value. No matter
-     * overflow happens or not, peer will detect it for us. */
-    if (aws_h2err_failed(aws_h2_stream_window_size_change(stream, (int32_t)window_update_size, true /*self*/))) {
-        stream->thread_data.window_size_self = INT32_MAX;
-    }
+    /* We already enqueued the window_update frame, just apply the change and let our peer check this value, no matter
+     * overflow happens or not. Peer will detect it for us. */
+    stream->thread_data.window_size_self += window_update_size;
```

Comment:
The largest legal value will be 3*(2^31-1), which is way less than INT64_MAX, so if the window_size_self overflows, there is no reason peer cannot find it out. So just apply the change and ignore the possible overflow. Or tell me that I am too naive.

Reply:
Oh no, it should be 2*(2^31-1). When the self-window-size is 2^31 -1, peer sends the data frame. Before we process the data frame, we send WINDOW_UPDATE frame to increment the window by 2^31 - 1. Now the peer stay in a valid state, but we can get  2*(2^31-1) in the self-window-size.